src/pyuvdata/uvdata/uvh5.py

```diff
@@ -449,6 +449,11 @@ def to_uvdata(
         )
         return uvd
 
+    @cached_property
+    def telescope(self):
+        """A Telescope object created from the data."""
+        return Telescope.from_hdf5(self, run_check=False)
```

Comment:
I think I'd prefer this go on the HDF5Meta class defined in `utils/io/hdf5.py` that FastUVH5Meta and FastCalH5Meta inherit from. That object already has some handling for telescope locations because it's needed for both the visibility data and calibrations solutions.

Reply:
It also looks like this isn't actually called in a test?

Reply:
I see, that makes sense. I've moved it there. And yes, I hadn't added a test. I have now.